dlt/common/typing.py

```diff
@@ -45,6 +45,11 @@
 
 from typing_extensions import is_typeddict as _is_typeddict
 
+if sys.version_info >= (3, 12):
```

Comment:
IMO it is safe to always import from typing_extensions. I do that ie. for Annotated. should we change this?